internal/k3d/create.go

```diff
@@ -196,3 +202,46 @@ func PostRunPrepareGitopsRepository(clusterName string,
 	}
 	return nil
 }
+
+func Restart(kubeconfig string) error {
+
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return fmt.Errorf("Unable to create config %w", err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("Error in creating clientset %w", err)
+	}
+
+	err = RestartDeployment(context.Background(), clientset, "argocd", "argocd-applicationset-controller")
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
Right, I guess my question is, is this the first time ever you connect to the kubernetes cluster in `k3d`? I find that difficult to grasp since by the point you want to restart a deployment, you already had to had connected to it to deploy it to begin with.

Reply:
Got it, so if this is the first time connecting to the cluster, who performs the initial deployment of ArgoCD?

Reply:
after we install argocd we deploy  to it using the argocd client package by providing the config and sorry patrick i missed a detail,we dont need to create clientset again ,its being created in the line [https://github.com/kubefirst/kubefirst/blob/e5cac33f599edcb6e4f4cce8e2f285254e5e0140/cmd/k3d/create.go#L820](url)

Reply:
i will remove this part and just call the RestartDeployment function

Reply:
No problem! Glad we could figure it out 😄